crates/aiken-lang/src/tipo/pattern.rs

```diff
@@ -467,7 +467,7 @@ impl<'a, 'b> PatternTyper<'a, 'b> {
                         let arguments = field_map
                             .fields
                             .iter()
-                            .sorted_by(|(a, _), (b, _)| a.cmp(b))
+                            .sorted_by(|(_, (a, _)), (_, (b, _))| a.cmp(b))
```

Comment:
FieldMap is defined as such:

```rust
pub struct FieldMap {
    pub arity: usize,
    pub fields: HashMap<String, (usize, Span)>,
    pub is_function: bool,
}
```

The `usize` is actually the field index in the definition.